src/web/debugger/simDriver.ts

```diff
@@ -0,0 +1,222 @@
+import { chooseWorkspaceAsync } from "../extension";
+import { buildProjectAsync } from "../makecodeOperations";
+import { Simulator } from "../simulator";
+import * as vscode from "vscode";
+import { PxtBreakpoint } from "./state";
+import { DebugProtocol } from "@vscode/debugprotocol";
+
+export enum SimulatorState {
+    Unloaded,
+    Stopped,
+    Pending,
+    Starting,
+    Running,
+    Paused,
+    Suspended
+}
+
+export enum SimulatorDebuggerCommand {
+    StepInto,
+    StepOver,
+    StepOut,
+    Resume,
+    Pause
+}
+
+interface SimulatorEventMap {
+    "breakpoint": pxsim.DebuggerBreakpointMessage;
+    "warning": pxsim.DebuggerWarningMessage;
+    "resume": undefined;
```

Comment:
I'm not sure I understand why this is undefined. If it was intentional, a comment could be helpful to explain the reasoning.

Reply:
it's only used for the typing below. I'll change it to never, which is a bit more self explanatory